functests/performance/topology_manager.go

```diff
@@ -0,0 +1,194 @@
+package performance
+
+import (
+	"fmt"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+var _ = Describe("[performance]Topology Manager", func() {
+	cs := testutils.NewClientSet("")
+	var workerRTNodes []corev1.Node
+
+	BeforeEach(func() {
+		var err error
+		workerRTNodes, err = nodes.GetByRole(cs, testutils.RoleWorkerRT)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(workerRTNodes).ToNot(BeEmpty())
+	})
+
+	It("should be enabled with the best-effort policy", func() {
+		kubeletConfig, err := nodes.GetKubeletConfig(cs, &workerRTNodes[0])
+		Expect(err).ToNot(HaveOccurred())
+
+		// verify topology manager feature gate
+		enabled, ok := kubeletConfig.FeatureGates[testutils.FeatureGateTopologyManager]
+		Expect(ok).To(BeTrue())
+		Expect(enabled).To(BeTrue())
+
+		// verify topology manager poicy
+		Expect(kubeletConfig.TopologyManagerPolicy).To(Equal(kubeletconfigv1beta1.BestEffortTopologyManagerPolicy))
+	})
+
+	Context("with the SR-IOV devices and static CPU's", func() {
+		var testpod *corev1.Pod
+		var sriovNode *corev1.Node
+
+		BeforeEach(func() {
+			sriovNodes := nodes.FilterByResource(cs, workerRTNodes, testutils.ResourceSRIOV)
+			if len(sriovNodes) < 1 {
```

Comment:
Don't we risk skipping if our deployment failed for some reason? Maybe it's better to fail (a bit like what you are doing above with worker-rt nodes). If we want to run the suite against a cluster without this feature, we can drive that through FOCUS / SKIP parameters)

Reply:
It still unclear how our baremetal CI will look, like will we have a different job for SR-IOV, or will we have a number of different environments under a single job, so we will need to check the existence of SR-IOV devices on tests runtime
Let's leave this check for the near future to ease run of all tests(locally), once we will have CI we can remove it.

Reply:
Ok, but please add a big TODO then :-)

Reply:
hh ok :)

Reply:
done